src/Maestro/maestro-angular/src/app/services/build-status.service.ts

```diff
@@ -10,9 +10,14 @@ export class BuildStatusService {
 
   public constructor(private http: HttpClient) { }
 
-  public getBranchStatus(account: string, project: string, definitionId: number, branch: string, count: number): Observable<BuildListResult> {
-    return this.http.get<BuildListResult>(`/_/AzDev/build/status/${account}/${project}/${definitionId}/${branch}?count=${count}`, {
+  public getBranchStatus(account: string, project: string, definitionId: number, branch: string, count: number, status: "inProgress"): Observable<BuildListResultInProgress>;
+  public getBranchStatus(account: string, project: string, definitionId: number, branch: string, count: number, status: "completed"): Observable<BuildListResultCompleted>;
+  public getBranchStatus(account: string, project: string, definitionId: number, branch: string, count: number, status: string): Observable<BuildListResultInProgress | BuildListResultCompleted>;
```

Comment:
Here we are returning `BuildListResultInProgress | BuildListResultCompleted` but you can call this method with `status` set to any other value in which case the result type won't match the actual data. There are two solutions - either we can remove this overload and only allow to call the method with `inProgress` or `completed` or we can return some generic `Build` object here.

I would choose the second way because that would allow you to get basic build info for any build irregardless of status without touching this service. In this case the build model would look something like:
```
export type BuildStatus = "completed" | "inProgress" /* ... | rest of the statuses */;

export interface Build {
  _links: { web: { href: string; } };
  id: number;
  status: BuildStatus;
}

export interface CompletedBuild extends Build {
  status: "completed";
  finishTime: string;
  result: "canceled" | "failed" | "none" | "partiallySucceeded" | "succeeded";
}

export interface BuildListResult<T extends Build> {
  count: number;
  value: T[];
}
```

and having that we can define API as:
```
public getBranchStatus(account: string, project: string, definitionId: number, branch: string, count: number, status: "completed"): Observable<BuildListResult<CompletedBuild>>;
public getBranchStatus(account: string, project: string, definitionId: number, branch: string, count: number, status: BuildStatus): Observable<BuildListResult<Build>>;
```